backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -78,3 +90,39 @@ class Meta:
                 name="unique_document_manager_profile_manager",
             ),
         ]
+
+
+# Function will be executed every time an instance of IndexManager is deleted.
+@receiver(post_delete, sender=IndexManager)
+def perform_vector_db_cleanup(sender, instance, **kwargs):
```

Comment:
What will happen if any error occurs in this function? Should we keep the entire logic in try except block to avoid unnecessary breakage of original flow. 

You might need to discuss this one with @hari-kuriakose or @arun-venkataswamy

Reply:
Moved everything inside the try block to avoid any disruption to other operations. @ritwik-g

Reply:
@gaya3-zipstack Got it. Thanks.